modules/apps/frontend-view-state/frontend-view-state-service/src/main/java/com/liferay/frontend/view/state/service/impl/FVSEntryServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.view.state.service.impl;
+
+import com.liferay.frontend.view.state.model.FVSEntry;
+import com.liferay.frontend.view.state.service.base.FVSEntryServiceBaseImpl;
+import com.liferay.portal.aop.AopService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.model.User;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Brian Wing Shun Chan
+ */
+@Component(
+	property = {
+		"json.web.service.context.name=fvs",
+		"json.web.service.context.path=FVSEntry"
+	},
+	service = AopService.class
+)
+public class FVSEntryServiceImpl extends FVSEntryServiceBaseImpl {
+
+	@Override
+	public FVSEntry addFVSEntry(String viewState) throws PortalException {
+		User user = getUser();
+
+		if ((user != null) && !user.isDefaultUser()) {
```

Comment:
As this is a remote service, should we check permissions before trying to add a FVSEntry?

Reply:
To copy my note from the opening message:
> The implementation of the remote service is the part of PR I have the least confidence in how to do correctly. The standard pattern for remote services in DXP is to check permissions with a PermissionChecker, however this requires a portlet, with a resource-actions/default.xml. FVS app does not have a portlet, it's just a service. What I did is added just a basic sanity check, that the user exists and is logged in. This may be a placeholder, util we do a full analysis and plan on what to do with permissions. Alternatively, we can just keep using local service for now

I have no idea how to do the standard permissions check. I believe this is impossible to do, as it requires a portlet. Maybe we can make a dummy portlet?

Reply:
I am not an expert in checking the permissions 😅 
We could create a dummy portlet for it, yes. Or do a little investigation to see if it is possible to check permissions without a portlet.